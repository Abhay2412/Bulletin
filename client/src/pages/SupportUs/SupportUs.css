.supportUsPage {
    display: flex;
    flex-direction: row;
}

.supportUs {
    padding-top: 5%;
    text-align: center;
    width: 80vw;
    justify-content: center;
    padding-left: 20%;
    padding-right: 20%;
    color: #68390D;
}

.supportDescription {
    padding-top: 5%;
}

.paymentButtons {
    background-color: #ECD1C5 !important;
}

.paymentButtonCluster {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    padding-left: 20%;
    padding-right: 20%;
    padding-top: 5%;
}

.donateButtonCluster {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 3%;
}



