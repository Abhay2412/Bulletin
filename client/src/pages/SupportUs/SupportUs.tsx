import { Typography, Button } from "@mui/material";
import React from "react";
import MockAccountSidebar from "../../components/MockAccountSidebar";
import paymentMethods from "../../imgs/paymentMethods.png";
import SideDrawer from "../../components/SideDrawer";
import "./SupportUs.css";

export default function SupportUs() {
  return (
    <>
      <SideDrawer />
      <div className="supportUsPage">
        <div className="supportUs">
          <Typography variant="h4" fontWeight={"bold"}>
            Support Us
          </Typography>

          <Typography variant="subtitle1" className="supportDescription">
            Like many Open Source Software projects, we rely heavily on the
            support and donations from users like yourself. If you found this
            software useful, or you received help from a volunteer/team member,
            please feel free to say thank you with a donation:
          </Typography>

          <div className="paymentButtonCluster">
            <Button className="paymentButtons">$5</Button>
            <Button className="paymentButtons">$10</Button>
            <Button className="paymentButtons">$15</Button>
            <Button className="paymentButtons">$50</Button>
            <Button className="paymentButtons">$100</Button>
          </div>
          <div className="donateButtonCluster">
            <Button className="paymentButtons">Other Amount</Button>

            <img src={paymentMethods} width="30%" />
          </div>
        </div>
      </div>
    </>
  );
}
