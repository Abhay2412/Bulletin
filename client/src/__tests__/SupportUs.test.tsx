import React from "react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import SupportUs from "../pages/SupportUs/SupportUs";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("testing <SupportUs /> component", () => {
  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <SupportUs />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
