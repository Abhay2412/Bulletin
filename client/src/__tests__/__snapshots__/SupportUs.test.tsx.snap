// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <SupportUs /> component should render properly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root makeStyles-drawerButton-9"
    disabled={false}
    id="side-drawer-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="ArrowForwardIosIcon"
        focusable="false"
        style={
          Object {
            "color": "#68390D",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
        />
      </svg>
    </span>
  </button>,
  <div
    className="supportUsPage"
  >
    <div
      className="supportUs"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-1313egv-MuiTypography-root"
      >
        Support Us
      </h4>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 supportDescription css-10wpov9-MuiTypography-root"
      >
        Like many Open Source Software projects, we rely heavily on the support and donations from users like yourself. If you found this software useful, or you received help from a volunteer/team member, please feel free to say thank you with a donation:
      </h6>
      <div
        className="paymentButtonCluster"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          $5
        </button>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          $10
        </button>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          $15
        </button>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          $50
        </button>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          $100
        </button>
      </div>
      <div
        className="donateButtonCluster"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root paymentButtons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Other Amount
        </button>
        <img
          src="paymentMethods.png"
          width="30%"
        />
      </div>
    </div>
  </div>,
]
`;
